### PR TITLE
fix(watch): watch will trigger when added new keys using `set`

### LINKED ISSUE #467: watch() with reactive() doesn't re-run when new properties are added to the reactive object
https://codesandbox.io/s/dry-leftpad-xrbtl

### PR DESCRIPTION
**Pull request by pikax**
fix #467

### CODE REVIEW

**Comment by antfu** [comment]
If I have recalled right, Vue 3 doesn't offer the `set` api. As the reactivity systems underneath are different, I think the `set` api is useful in some situations with Vue 2. Should we add a note in the readme to mention this difference?

**Comment by pikax** [comment]
`set` is only to circumvent the limitation on adding a new `key` limitation on `vue2`. In `vue3` there's no such limitation when using proxys.

We should add it to the readme.

File: README.md [hunk 1]
```diff
@@ -174,6 +174,27 @@ a.list.push(
...
 ```
 
+</details>
+
+<details1>
+<summary>
+⚠️ adding new reactive properties `Vue.set`
+</summary>
+
+> Part of the limitation of [Vue 2.x reactivity](https://vuejs.org/v2/guide/reactivity.html#For-Objects) is to track new keys on an `object` you need to call `Vue.set`.
```
Comment (antfu) [inline README.md:184]:
```suggestion
<summary>
⚠️ `set` workaround for adding new reactive properties
</summary>

> ⚠️ Warning: `set` does NOT exist in Vue 3. We provide it as a workaround here, due to the limitation of [Vue 2.x reactivity system](https://vuejs.org/v2/guide/reactivity.html#For-Objects). In Vue 2, you will need to call `set` to track new keys on an `object`(similar to `Vue.set` but for `reactive objects` created by the Composition API). In Vue 3, you can just assign them like normal objects.
```

File: README.md [hunk 2]
```diff
@@ -174,6 +174,27 @@ a.list.push(
 a.list[1].count === 1 // true
 ```
 
+</details>
+
+<details1>
```
Comment (antfu) [inline README.md:179]:
```suggestion
<details>
```
😆

**Pull request merged**